user/tests/wiring/no_fixture/interrupts.cpp

```diff
@@ -0,0 +1,37 @@
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+test(interrupts_atomic_section)
+{
+	// using SysTick for convenience - we could use a stm32 timer interrupt for a more generic test
+	// millis() will not increment while interrupts are disabled.
+	// micros() does increment
+
+	uint32_t start_micros = micros();
+	uint32_t start_millis, end_millis;
+	{
+		ATOMIC_SECTION();
+		start_millis = millis();
+		while (micros()-start_micros<2000)
+		{
+			// validates that atomic sections can be nested and interrupts restored when the outermost one exits
+			ATOMIC_SECTION();
```

Comment:
I'm thinking `HAL_enable_irq(prev);` will be executed in the while() expression test, but now that I look a bit closer since `prev` was created inside the while() loop it will already have the interrupts masked, therefor it will not re-enable them by accident.